### DISCUSSION TITLE
Azure application gateway 403 for creating roles

### DISCUSSION THREAD

**Discussion by srivishnuk** [strapi/strapi#7947] [Legacy Topics]
Hello All,

Azure application gateway is blocking role creation on Strapi, Throw's error 403 and reports RFI attack, Attached is screenshot

![image](https://user-images.githubusercontent.com/7478337/93584861-8a276d00-f9c3-11ea-819d-72682dfbf7e0.png)

**Comment by derrickmehaffy** [comment]
Not sure I understand where this is coming from?

**Reply by srivishnuk** [reply]
Hello Derrick,

This is from Azure Application Gateway WAF

**Reply by derrickmehaffy** [reply]
I'm not sure I understand why it's thinking there is a RFI attack, did you set the `./config/server.js` or `./config/env/*/server.js` url key to whatever your public URL is?

**Reply by srivishnuk** [reply]
@derrickmehaffy  domain name is pointing to Azure application gateway cname , and strapi is installed on a VM on Azure which is Backend to Azure Application gateway with private IP and Port , config is ./config/server.js it has private IP and port as access URL everything works fine except for adding roles which shows RFI attack in WAF logs

**Reply by derrickmehaffy** [reply]
I'm not even sure how to begin to reproduce this, we will need some information to be able to reproduce and track down the issue :/

**Reply by srivishnuk** [reply]
@derrickmehaffy  i can give you access to the current test setup environment and its details to your mail. Please confirm

**Reply by derrickmehaffy** [reply]
Sure that's fine, send them to derrick.mehaffy@strapi.io

**Reply by srivishnuk** [reply]
Thanks @derrickmehaffy  invited you to Azure and also sent you a email with subject  **Strapi RFI Azure** which has strapi login credentials and steps to reproduce

**Reply by derrickmehaffy** [reply]
Great thank you, I'll try and review this sometime this week (lots going on ;) )

**Reply by srivishnuk** [reply]
Welcome @derrickmehaffy , Thank you for checking. Understand , will wait :)

**Reply by derrickmehaffy** [reply]
@srivishnuk as I mentioned in my email, this is something entirely unrelated to Strapi. There are some very strict rules that need to be excluded and I have very little exp with the WAF in Azure (was a pain in the butt to even get the logs :laughing: ). 

I would suggest in this case you reach out to Azure to see if they can help you add exclusions.
